internal/slicer/slicer.go

```diff
@@ -242,17 +254,28 @@ func Run(options *RunOptions) (*Report, error) {
 			if err != nil {
 				return nil, err
 			}
-			err = report.Add(slice, entry)
-			if err != nil {
-				return nil, err
+
+			// Do not add paths with "until: mutate".
+			if pathInfo.Until != setup.UntilMutate {
+				err = report.Add(slice, entry)
+				if err != nil {
+					return nil, err
+				}
 			}
 		}
 	}
 
 	// Run mutation scripts. Order is fundamental here as
 	// dependencies must run before dependents.
 	checkWrite := func(path string) error {
-		if !pathInfos[path].Mutable {
+		pi, ok := pathInfos[path]
+		if !ok {
```

Comment:
We don't need this test since the zero value should work okay. Actually, I suspect none of the changes in this function are required, since the original message tells exactly what is expected for the function call to work. For example, this:

> cannot write to directory: /foo/bar/

... makes it look like you have permission issues on that directory. While this:

> cannot write file which is not mutable: /foo/bar/

... says this was supposed to be a file, mutable.

Similarly, "unlisted", doesn't say unlisted where. While the second message above says it: you must mark this as a mutable file.  So let's please just revert the change to the function, assuming it works fine of course.

Reply:
When I read the existing error message I was a little bit confused that it was telling me that the "file was not mutable" but then pointing me to a directory. Hence, why I changed it. If I read it as "only mutable files can be written to", then it is clear to me.

I understand this might be a problem only on my side so I have reverted the code and removed the extra error reporting.